### PR TITLE
Handled the twice button click while text to Pdf

### LINKED ISSUE #729: [ PDF CONVERTER  version 8.6.5]  Text to Pdf button is not working properly
#### Project Information
* Repository    :  https://github.com/Swati4star/Images-to-PDF
* Project Name  :  PDF CONVERTER
* Publisher     : Swati Garg

#### Expected behavior

I Expected that when I use the text to pdf option  in this  application the text document  should be converted into PDF file 

#### Actual behavior

The  Text to Pdf button is not working properly  while i convert the file text to pdf error come  you can see in bug video properly 

#### How to reproduce
* [Download ](https://play.google.com/store/apps/details?id=swati4star.createpdf)letest version and install it 
* open the application and click on Text To PDF  option
* select your text file 
* click on create PDF
* enter the  new file name  
* Note the Bug
* Browser/App version:   8.6.5
* Operating system:  8.0.0

#### Recording Of The Bug
https://youtu.be/3oLsUD_2tXg
#### GitHub Account
https://github.com/XAMIR555

# logcat
[logcat.txt](https://github.com/Swati4star/Images-to-PDF/files/3103498/logcat.txt)

### PR DESCRIPTION
**Pull request by jddeep**
# Description

Handled the twice button click while text to Pdf

Fixes #729 

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jddeep** [comment]
@Swati4star @vyankatesh24 Please review.

**Comment by jddeep** [comment]
@Swati4star Yes and i could not get why? I mean those changes were merged to master branch then why are those changes again showing up here?

**Comment by Swati4star** [comment]
Try rebasing your branch with master. I think that could be the problem

**Comment by jddeep** [comment]
@Swati4star Done. Please review.

**Comment by Swati4star** [comment]
what exactly was the issue?

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 1]
```diff
@@ -357,6 +357,9 @@ private void showFontSize() {
 
     @OnClick(R.id.createtextpdf)
     public void openCreateTextPdf() {
+        if (!mPermissionGranted)
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: we are already checking for permissions in `selectTextFile` function

**Comment by jddeep** [comment]
@Swati4star the issue was that: The Text to Pdf button after selecting file needed to be clicked twice. This was after requesting for perms it would `return` back. Thus that needed to be changed.

**Comment by Swati4star** [comment]
@vyankatesh24 Can you verify the fix once please?

**Comment by vyankatesh24** [comment]
> @Swati4star the issue was that: The Text to Pdf button after selecting file needed to be clicked twice. This was because after requesting for perms it would `return` back. Thus that needed to be changed.

No @jddeep this is not the issue. 
Have you watched the video uploaded by @XAMIR555
In that video I am not seeing this issue.

**Comment by jddeep** [comment]
@vyankatesh24 Well the issue was so obvious that I believed this would have been it. But now on seeing the video it seems that the file is not getting loaded properly. Well @XAMIR555 can you post your logcat? @vyankatesh24 What do you say?

**Comment by vyankatesh24** [comment]
> @vyankatesh24 Well the issue was so obvious that I believed this would have been it. But now on seeing the video it seems that the file is not getting loaded properly. Well @XAMIR555 can you post your logcat? @vyankatesh24 What do you say?

@jddeep log has been already provided. See issue description properly.

**Comment by jddeep** [comment]
@vyankatesh24 Yeah but i can't track the problem from the txt file. There are many exceptions in the txt file. A screenshot would have done good.

**Comment by jddeep** [comment]
@Swati4star I rightly agree that #729 is device specific. But this PR does solve the issue for twice clicking the button which is in general with other devices. I and @vyankatesh24 have solved this issue with other fragments also in this way.

**Comment by vyankatesh24** [comment]
> @Swati4star I rightly agree that #729 is device specific. But this PR does solve the issue for twice clicking the button which is in general with other devices. I and @vyankatesh24 have solved this issue with other fragments also in this way.

@jddeep But in this fragment the "double click need"  issue is not present.

**Comment by jddeep** [comment]
@vyankatesh24 But this was present when I started working on it. Hence I became surer that this would have been the issue.

**Comment by vyankatesh24** [comment]
That issue has already been solved in #665.

**Comment by jddeep** [comment]
But I experienced it. But as you are telling otherwise then i guess i will leave this. @vyankatesh24 Please see then what can be done in this PR.

**Comment by Swati4star** [comment]
@vyankatesh24 If it is already fixed, then we shouldn't merge this PR?

**Comment by vyankatesh24** [comment]
> @vyankatesh24 If it is already fixed, then we shouldn't merge this PR?

Yes @Swati4star

**Pull request closed without merging**